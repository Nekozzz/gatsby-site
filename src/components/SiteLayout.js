/**
 * Layout component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */

import React from "react"
import PropTypes from "prop-types"
import {useStaticQuery, graphql, Link} from "gatsby"
import Header from "./header"
import 'antd/dist/antd.css'
import "./SiteLayout.css"
import {Layout, Menu} from "antd";
const { Sider, Content } = Layout;


const SiteLayout = ({ children, selectedMenuItem = 'main'}) => {
  const data = useStaticQuery(graphql`
    query Asm {
      site {
        siteMetadata {
          title
        }
      }
      hasura {
        car_brands {
          slug
          brand_name
        }
      }
    }
  `);

  /*  return (
      <>
        <Header siteTitle={data.site.siteMetadata.title} />
        <div
          style={{
            margin: `0 auto`,
            maxWidth: 960,
            padding: `0 1.0875rem 1.45rem`,
          }}
        >
          <main>{children}</main>
          <footer>
            © {new Date().getFullYear()}, Built with
            {` `}
            <a href="https://www.gatsbyjs.org">Gatsby</a>
          </footer>
        </div>
      </>
    )
  };*/

  return (
    <>
      <Header siteTitle={data.site.siteMetadata.title}/>
      <main className={'main-container'}>
        <Layout>
          <Sider theme={'light'}>
            <Menu mode="inline" selectedKeys={selectedMenuItem}>
              <Menu.Item key={'main'}>
                <Link to={'/'}>
                  Главная страница
                </Link>
              </Menu.Item>
              {
                data.hasura.car_brands.map((carBrand) => (
                  <Menu.Item key={carBrand.slug} style={{
                      'display': 'flex',
                      'flexDirection': 'column',
                    }}>
                    <Link to={'/'+carBrand.slug}>
                      {carBrand.brand_name}
                    </Link>
                  </Menu.Item>
                ))
              }

            </Menu>
          </Sider>

          <Layout className="site-layout">
            <Content
              style={{
                margin: '24px 16px',
                padding: 24,
                minHeight: 280,
              }}
            >
              {children}
            </Content>
          </Layout>
        </Layout>
      </main>

      <footer>
        © {new Date().getFullYear()}, Built with
        {` `}
        <a href="https://www.gatsbyjs.org">Gatsby</a>
      </footer>
    </>
  );
}


SiteLayout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default SiteLayout;
