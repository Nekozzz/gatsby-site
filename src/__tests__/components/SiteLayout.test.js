import React from "react"
import { useStaticQuery } from "gatsby"
import renderer from "react-test-renderer"
import SiteLayout from "../../components/SiteLayout";
import * as Gatsby from "gatsby";

beforeEach(() => {
  const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
  useStaticQuery.mockImplementationOnce(() => (
    {
      site: {
        siteMetadata: {
          title: 'asd'
        }
      },
      hasura: {
        car_brands: [
          {
            slug: 'adidas',
            brand_name: 'Adiads'
          },
          {
            slug: 'cat',
            brand_name: 'Cat man'
          },
        ]
      }
    }
  ))
})

describe("SiteLayout", () => {
  test("renders correctly", () => {

    // const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
    // useStaticQuery.mockImplementation(() => ({
    //   site: {
    //     siteMetadata:{
    //       title: 'asd'
    //     }
    //   },
    //   hasura: {
    //     car_brands: [
    //       {
    //         slug: 'adidas',
    //         brand_name: 'Adiads'
    //       },
    //       {
    //         slug: 'cat',
    //         brand_name: 'Cat man'
    //       },
    //     ]
    //   }
    // }));


    const tree = renderer
      .create(
        <SiteLayout>content</SiteLayout>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})