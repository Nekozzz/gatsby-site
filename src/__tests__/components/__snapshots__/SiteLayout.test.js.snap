// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteLayout renders correctly 1`] = `
Array [
  <header
    style={
      Object {
        "background": "rebeccapurple",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          asd
        </a>
      </h1>
    </div>
  </header>,
  <main
    className="main-container"
  >
    <section
      className="ant-layout ant-layout-has-sider"
    >
      <aside
        className="ant-layout-sider ant-layout-sider-light"
        style={
          Object {
            "flex": "0 0 200px",
            "maxWidth": "200px",
            "minWidth": "200px",
            "width": "200px",
          }
        }
      >
        <div
          className="ant-layout-sider-children"
        >
          <ul
            className="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
            onMouseEnter={[Function]}
            onTransitionEnd={[Function]}
            role="menu"
            style={Object {}}
          >
            <li
              className="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="menuitem"
              style={
                Object {
                  "paddingLeft": 24,
                }
              }
            >
              <a
                href="/"
              >
                Главная страница
              </a>
            </li>
            <li
              className="ant-menu-item ant-menu-item-only-child"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="menuitem"
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "paddingLeft": 24,
                }
              }
            >
              <a
                href="/adidas"
              >
                Adiads
              </a>
            </li>
            <li
              className="ant-menu-item ant-menu-item-only-child"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="menuitem"
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "paddingLeft": 24,
                }
              }
            >
              <a
                href="/cat"
              >
                Cat man
              </a>
            </li>
          </ul>
        </div>
      </aside>
      <section
        className="ant-layout site-layout"
      >
        <main
          className="ant-layout-content"
          style={
            Object {
              "margin": "24px 16px",
              "minHeight": 280,
              "padding": 24,
            }
          }
        >
          content
        </main>
      </section>
    </section>
  </main>,
  <footer>
    © 
    2020
    , Built with
     
    <a
      href="https://www.gatsbyjs.org"
    >
      Gatsby
    </a>
  </footer>,
]
`;
